samples/04.0-file-transfer/integration-tests/src/test/java/org/eclipse/dataspaceconnector/extension/sample/test/FileTransferSampleTest.java

```diff
@@ -0,0 +1,241 @@
+        await().atMost(TIMEOUT).pollInterval(POLL_INTERVAL).untilAsserted(() -> {
+                    var result = RestAssured
+                        .given()
+                            .headers(API_KEY_HEADER_KEY, API_KEY_HEADER_VALUE)
+                        .when()
+                            .get(LOOK_UP_CONTRACT_AGREEMENT_URI, contractNegotiationId)
+                        .then()
+                            .statusCode(HttpStatus.SC_OK)
+                            .body("state", equalTo("CONFIRMED"))
+                            .body("contractAgreementId", not(emptyString()))
+                            .extract().body().jsonPath().getString("contractAgreementId");
+
+                    localContractAgreementId.set(result);
+
+                }
+        );
+
+        contractAgreementId = localContractAgreementId.get();
+    }
+
+    /**
+     * Assert that a POST request to initiate transfer process is successful.
+     * This methods corresponds to the command in the sample: curl -X POST -H "Content-Type: application/json" -H "X-Api-Key: password" -d @samples/04.0-file-transfer/filetransfer.json "http://localhost:9192/api/v1/data/transferprocess"
+     * @throws IOException Thrown if there was an error accessing the transfer request file defined in {@link FileTransferSampleTest#TRANSFER_FILE_PATH}.
+     */
+    void assertRequestFile() throws IOException {
+        File transferJsonFile = getFileFromRelativePath(TRANSFER_FILE_PATH);
+        DataRequest sampleDataRequest = readAndUpdateTransferJsonFile(transferJsonFile, contractAgreementId);
+
+        JsonPath jsonPath = RestAssured
+            .given()
+                .headers(API_KEY_HEADER_KEY, API_KEY_HEADER_VALUE)
+                .contentType(ContentType.JSON)
+                .body(sampleDataRequest)
+            .when()
+                .post(INITIATE_TRANSFER_PROCESS_URI)
+            .then()
+                .statusCode(HttpStatus.SC_OK)
+                .body("id", not(emptyString()))
+                .extract()
+                .jsonPath();
+
+        String transferProcessId = jsonPath.get("id");
+
+        assertThat(transferProcessId).isNotEmpty();
+    }
+
+    /**
+     * Reads a transfer request file with changed value for contract agreement ID and file destination path.
+     * @param transferJsonFile A {@link File} instance pointing to a JSON transfer request file.
+     * @param contractAgreementId This string containing a UUID will be used as value for the contract agreement ID.
+     * @return An instance of {@link DataRequest} with changed values for contract agreement ID and file destination path.
+     * @throws IOException Thrown if there was an error accessing the file given in transferJsonFile.
+     */
+    DataRequest readAndUpdateTransferJsonFile(File transferJsonFile, String contractAgreementId) throws IOException {
+        // create object mapper instance
+        ObjectMapper mapper = new ObjectMapper();
```

Comment:
could be static final field

Reply:
Do you mean to promote `ObjectMapper mapper = new ObjectMapper()` as a field and make it static final?
It's used only once in this method. What would be the benefit?

However, I have changed the method to be static as it has no dependencies to the instance.

Reply:
Correct. The intention is to make the UT code more compact and focus on what is really tested, therefore more readable

Reply:
Makes sense. Changed it.